### ISSUE TITLE
Invalid types in angular project

### ISSUE THREAD

**Issue by KiraL91** [xeokit/xeokit-sdk#832]
**Unable to build Angular project**
I tryed to build a basic Angular project that imports the xeokit package. The console shows a few errors in 2.2.5-beta-2

**To Reproduce**

1. npm i @xeokit/xeokit-sdk@2.2.5-beta-2
2. Create new Angular component
3. Import package in component. i.e. `import { Viewer} from '@xeokit/xeokit-sdk';`
4. ng serve

**Erros**
```bash

./node_modules/@xeokit/xeokit-sdk/dist/xeokit-sdk.es.js:98332:45-55 - Warning: Critical dependency: the request of a dependency is an expression

./node_modules/@xeokit/xeokit-sdk/dist/xeokit-sdk.es.js:162596:14-45 - Warning: Module not found: Error: Can't resolve 'asciify-image' in 'C:/path/to/project\node_modules\@xeokit\xeokit-sdk\dist'

Warning: C:/path/to/project\src\app\components\viewer-xeokit\viewer-xeokit.component.ts depends on '@xeokit/xeokit-sdk'. CommonJS or AMD dependencies can cause optimization bailouts.      
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies



Error: node_modules/@xeokit/xeokit-sdk/types/plugins/AnnotationsPlugin/AnnotationsPlugin.d.ts:1:28 - error TS2307: Cannot find module 'viewer/scene/webgl/PickResult' or its corresponding type declarations.

1 import { PickResult } from "viewer/scene/webgl/PickResult";
                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/@xeokit/xeokit-sdk/types/plugins/GLTFLoaderPlugin/GLTFLoaderPlugin.d.ts:11:53 - error TS7010: 'Function', which lacks return-type annotation, implicitly has an 'any' return type.

11   getMetaModel(metaModelSrc: string | number, ok: { Function(); }, error: { Function();}): void;
                                                       ~~~~~~~~~~~


Error: node_modules/@xeokit/xeokit-sdk/types/plugins/GLTFLoaderPlugin/GLTFLoaderPlugin.d.ts:11:77 - error TS7010: 'Function', which lacks return-type annotation, implicitly has an 'any' return type.

11   getMetaModel(metaModelSrc: string | number, ok: { Function(); }, error: { Function();}): void;
                                                                               ~~~~~~~~~~~


Error: node_modules/@xeokit/xeokit-sdk/types/plugins/STLLoaderPlugin/index.d.ts:1:15 - error TS2307: Cannot find module './STLDefaultDataSource.js' or its corresponding type declarations.

1 export * from "./STLDefaultDataSource.js";
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/@xeokit/xeokit-sdk/types/viewer/Viewer.d.ts:169:3 - error TS7010: 'addPlugin', which lacks return-type annotation, implicitly has an 'any' return type.

169   addPlugin(plugin: Plugin);
      ~~~~~~~~~


Error: node_modules/@xeokit/xeokit-sdk/types/viewer/metadata/index.d.ts:1:15 - error TS1149: File name 'C:/path/to/project/node_modules/@xeokit/xeokit-sdk/types/viewer/metadata/Metamodel.d.ts' differs from already included file name 'C:/path/to/project/node_modules/@xeokit/xeokit-sdk/types/viewer/metadata/MetaModel.d.ts' only in casing.
  The file is in the program because:
    Imported via "./MetaModel" from file 'C:/path/to/project/node_modules/@xeokit/xeokit-sdk/types/viewer/metadata/MetaScene.d.ts' with packageId '@xeokit/xeokit-sdk/types/viewer/metadata/MetaModel.d.ts@2.2.5-beta-2'
    Imported via "./MetaModel" from file 'C:/path/to/project/node_modules/@xeokit/xeokit-sdk/types/viewer/metadata/MetaObject.d.ts' with packageId '@xeokit/xeokit-sdk/types/viewer/metadata/MetaModel.d.ts@2.2.5-beta-2'
    Imported via "./Metamodel" from file 'C:/path/to/project/node_modules/@xeokit/xeokit-sdk/types/viewer/metadata/index.d.ts' with packageId '@xeokit/xeokit-sdk/types/viewer/metadata/Metamodel.d.ts@2.2.5-beta-2'

1 export * from "./Metamodel";
                ~~~~~~~~~~~~~

  node_modules/@xeokit/xeokit-sdk/types/viewer/metadata/MetaScene.d.ts:3:27
    3 import { MetaModel } from "./MetaModel";
                                ~~~~~~~~~~~~~
    File is included via import here.
  node_modules/@xeokit/xeokit-sdk/types/viewer/metadata/MetaObject.d.ts:1:27
    1 import { MetaModel } from "./MetaModel";
                                ~~~~~~~~~~~~~
    File is included via import here.


Error: node_modules/@xeokit/xeokit-sdk/types/viewer/scene/materials/Fresnel.d.ts:149:29 - error TS2307: Cannot find module '../webgl/RenderState.js' or its corresponding type declarations.        

149 import { RenderState } from "../webgl/RenderState.js";
                                ~~~~~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/@xeokit/xeokit-sdk/types/viewer/scene/materials/Texture.d.ts:261:29 - error TS2307: Cannot find module '../webgl/RenderState.js' or its corresponding type declarations.        

261 import { RenderState } from "../webgl/RenderState.js";
                                ~~~~~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/@xeokit/xeokit-sdk/types/viewer/scene/math/math.d.ts:27:3 - error TS7010: 'getAABB2Center', which lacks return-type annotation, implicitly has an 'any' return type.

27   getAABB2Center(aabb: number[], dest?: (values?: number[])=> number[]);
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/@xeokit/xeokit-sdk/types/viewer/scene/math/math.d.ts:33:3 - error TS7010: 'getAABB3Center', which lacks return-type annotation, implicitly has an 'any' return type.

33   getAABB3Center(aabb: number[], dest?: (values?: number[])=> number[]);
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/@xeokit/xeokit-sdk/types/viewer/scene/scene/Scene.d.ts:2:24 - error TS1149: File name 'C:/path/to/project/node_modules/@xeokit/xeokit-sdk/types/viewer/scene/camera/Camera.d.ts' differs from already included file name 'C:/path/to/project/node_modules/@xeokit/xeokit-sdk/types/viewer/scene/Camera/Camera.d.ts' only in casing.
  The file is in the program because:
    Imported via "./Camera" from file 'C:/path/to/project/node_modules/@xeokit/xeokit-sdk/types/viewer/scene/Camera/index.d.ts' with packageId '@xeokit/xeokit-sdk/types/viewer/scene/Camera/Camera.d.ts@2.2.5-beta-2'
    Imported via "../camera/Camera" from file 'C:/path/to/project/node_modules/@xeokit/xeokit-sdk/types/viewer/scene/scene/Scene.d.ts' with packageId '@xeokit/xeokit-sdk/types/viewer/scene/camera/Camera.d.ts@2.2.5-beta-2'

2 import { Camera } from "../camera/Camera";
                         ~~~~~~~~~~~~~~~~~~

  node_modules/@xeokit/xeokit-sdk/types/viewer/scene/Camera/index.d.ts:1:15
    1 export * from "./Camera";
                    ~~~~~~~~~~
    File is included via import here.


Error: node_modules/@xeokit/xeokit-sdk/types/viewer/scene/scene/Scene.d.ts:525:3 - error TS7010: 'render', which lacks return-type annotation, implicitly has an 'any' return type.

525   render(forceRender?: boolean);
      ~~~~~~


Error: node_modules/@xeokit/xeokit-sdk/types/viewer/scene/scene/Scene.d.ts:721:30 - error TS7006: Parameter 'callback' implicitly has an 'any' type.

721   withObjects(ids: string[], callback): boolean;
                                 ~~~~~~~~


Error: node_modules/@xeokit/xeokit-sdk/types/viewer/scene/scene/Scene.d.ts:726:3 - error TS7010: 'destroy', which lacks return-type annotation, implicitly has an 'any' return type.

726   destroy(); void;
      ~~~~~~~


Error: node_modules/@xeokit/xeokit-sdk/types/viewer/scene/scene/Scene.d.ts:726:14 - error TS7008: Member 'void' implicitly has an 'any' type.

726   destroy(); void;
```

**Comment by mlankamp** [comment]
Thank you for the feedback on the typings. I'm going to have a look at the different problems.

**Comment by KiraL91** [comment]
I was able to build the project by resolving some minor bugs (i.e. the return type of some methods was missing). However, there are some missing in different classes, such as in ```NavCubePluginConfiguration.d.ts``` that I encountered while executing the examples.

```javascript
export declare type NavCubePluginConfiguration = {
  ...
   //  TODO: Add additional parameters

  synchProjection?: boolean;
  size?: number;
  alignment?: string;
  bottomMargin?: number;
  rightMargin?: number;
}
```
I suppose that there are other problems that need to be fixed, so I'll try to update this open issue with the detected bugs.

**Comment by mlankamp** [comment]
As far as I can see, these properties don't exist anymore and has to do with some out-dated examples. 

The typescript typings a created from the source (
https://github.com/xeokit/xeokit-sdk/blob/b001d5bb13546b3716be0ff191b4e5247c724e87/src/plugins/NavCubePlugin/NavCubePlugin.js#L98), there you can see the properties don't exist.

The missing return types etc have been fixed, just waiting on a PR and a new package to be published.

**Comment by KiraL91** [comment]
Hi,
I encountered some problems while executing the [HoverShowLabels](https://github.com/xeokit/xeokit-sdk/blob/master/examples/annotations_hoverShowLabels.html) example.

```js
...
    annotations.on("markerMouseEnter", (annotation) => {
      annotation.setLabelShown(true);
    });

    annotations.on("markerMouseLeave", (annotation) => {
      annotation.setLabelShown(false);
    });
```
The **two** parameters are marked as invalid. I think that in `AnnotationsPlugin.d.ts`:

```js
  /**
   * Fires when a annotation is created.
   * @param {String} event The annotationCreated event
   * @param {Function} callback Callback fired on the event
   */
   on(event: "annotationCreated", callback: (annotationId: string)=> void): void;

//  Should be something similar to:
   on(event: string, callback: (annotation: Annotation)=> void): void;

   /**
    * Fires when a annotation is destroyed.
    * @param {String} event The annotationDestroyed event
    * @param {Function} callback Callback fired on the event
    */
   on(event: "annotationDestroyed", callback: (annotationId: string)=> void): void;

// Remove second method?
```

In the same example

```js 
annotations.createAnnotation({
          id: "myAnnotation5",
          worldPos: [8.947, 4.257, 9.297],
          occludable: true,
          markerShown: true,
          labelShown: false,

          values: {
            glyph: "A5",
            title: "The East upstairs wall",
            description: "The East-facing upstairs wall",
            markerBGColor: "orange"
          }
        });
```

The field value is marked as invalid:

```js
createAnnotation(params: {
...
values?: {
          String: (string | number);
      };
...
}
```


And also
```js
           onMarkerClick: (annotation) => {
             annotation.setLabelShown(!annotation.getLabelShown());
           }
```

onMarkerClick is not defined. Could you give me some insigths on how to use this function?

Thank you

**Comment by mlankamp** [comment]
hi,

I've made a new PR with the newest typings:
* events `markerMouseLeave`, `markerMouseLeave` and `markerClicked` are added 
* updated the `values` signature

the onMarkerClick is unknown, I think it should be used as:
```typescript
annotations.on("markerClicked", (annotation) => {
  annotation.setLabelShown(!annotation.getLabelShown());
});
```
